modules/siddhi-core/src/main/java/io/siddhi/core/util/cache/CacheExpirer.java

```diff
@@ -0,0 +1,226 @@
+                                        getOutputAttributesForCaching());
+                        clearCacheAndReload(loadedDataFromStore);
+                        storeTable.setCacheLastReloadTime(siddhiAppContext.getTimestampGenerator().currentTime());
+                    }
+                } finally {
+                    AbstractQueryableRecordTable.queryStoreWithoutCheckingCache.set(Boolean.FALSE);
+                }
+            } else {
+                cacheTable.delete(generateDeleteEventChunk(), cacheExpiryCompiledCondition);
+            }
+            } catch (ConnectionUnavailableException e) {
+                throw new SiddhiAppRuntimeException(siddhiAppContext.getName() + ": " + e.getMessage());
+            }
+        } else {
+            readWriteLock.writeLock().lock();
+            try {
+                AbstractQueryableRecordTable.queryStoreWithoutCheckingCache.set(Boolean.TRUE);
+                try {
+                    loadedDataFromStore = storeTable.query(stateEventForCaching,
+                            storeTable.getCompiledConditionForCaching(),
+                            storeTable.getCompiledSelectionForCaching(), storeTable.getOutputAttributesForCaching());
+                    storeTable.setStoreTableSize(findEventChunkSize(loadedDataFromStore));
+                    storeTable.setStoreSizeLastCheckedTime(siddhiAppContext.getTimestampGenerator().currentTime());
+                } finally {
+                    AbstractQueryableRecordTable.queryStoreWithoutCheckingCache.set(Boolean.FALSE);
+                }
+                if (storeTable.getStoreTableSize() <= storeTable.getMaxCacheSize()) {
+                    if (storeTable.getCacheLastReloadTime() < siddhiAppContext.getTimestampGenerator().currentTime() +
+                            retentionPeriod) {
+                        clearCacheAndReload(loadedDataFromStore);
+                        storeTable.setCacheLastReloadTime(siddhiAppContext.getTimestampGenerator().currentTime());
+                    }
+                } else {
+                    cacheTable.delete(generateDeleteEventChunk(), cacheExpiryCompiledCondition);
+                }
+            } catch (Exception e) {
+                throw new SiddhiAppRuntimeException(siddhiAppContext.getName() + ": " + e.getMessage());
+            } finally {
+                readWriteLock.writeLock().unlock();
+            }
+        }
+        log.debug(siddhiAppContext.getName() + ": CacheExpirer ended");
+    }
+
+    private void clearCacheAndReload(StreamEvent loadedDataFromStore) {
+        readWriteLock.writeLock().lock();
+        try {
+            ((CacheTable) cacheTable).deleteAll();
+            if (loadedDataFromStore != null) {
+                ((CacheTable) cacheTable).addStreamEventUptoMaxSize(loadedDataFromStore);
+            }
+        } finally {
+            readWriteLock.writeLock().unlock();
+        }
+    }
+
```

Comment:
Unnecessary new lines

Reply:
Fixed in https://github.com/siddhi-io/siddhi/pull/1377